pkg/logging/scoped.go

```diff
@@ -0,0 +1,81 @@
+package logging
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+)
+
+type loggerRegistry struct {
+	sync.RWMutex
+	loggers     map[string]*LeveledLogger
+	scopeLevels map[string]LogLevel
+}
+
+var (
+	registry = &loggerRegistry{
+		loggers:     make(map[string]*LeveledLogger),
+		scopeLevels: make(map[string]LogLevel),
+	}
+)
+
+// NewScopedLogger returns a predefined logger for the given logging scope
+// NB: Can be used idempotently
+func NewScopedLogger(scope string) *LeveledLogger {
+	registry.Lock()
+	defer registry.Unlock()
+
+	scope = strings.ToLower(scope)
+	if _, found := registry.loggers[scope]; !found {
+		registry.loggers[scope] = NewLeveledLoggerForScope(scope)
+
+		// Handle a logger being created after init() is run
+		level := LogLevelDisabled
+		if allLevel, found := registry.scopeLevels["all"]; found {
+			level = allLevel
+		}
+		if scopeLevel, found := registry.scopeLevels[scope]; found {
+			if scopeLevel > level {
+				level = scopeLevel
+			}
+		}
+		if level > LogLevelDisabled {
+			registry.loggers[scope].SetLevel(level)
+		}
+	}
+	return registry.loggers[scope]
+}
+
+func init() {
```

Comment:
@mjmac @backkem 

What do you think of using the settings engine to init this instead? https://github.com/pions/webrtc/blob/master/settingengine.go

I see this as the our way of setting `non-standard WebRTC things`, also makes it easier to demo in our `examples/` I would have no problem parsing args/env variables in the examples themselves though

Reply:
one pro-environment-variable argument is this: Gstreamer (and glib) do it, so people used to working with multimedia packages from these libraries will expect environment variables (or at least it's a sideways move)

As long as it's well documented...

Reply:
I don't love it, to be honest. It doesn't feel very ergonomic from a developer perspective. Environment variables have a long history of use for this sort of thing on unix-like platforms. It would be really annoying to me to have to fiddle with a configuration file every time I wanted to tweak logging while I was hacking on something.

That having been said, I'm open to the idea of a compromise that enables both use cases. I don't mind making the internals a bit more complicated in order to make life easier for users and developers.

How would the settings engine work for this? If you guys would propose examples of the way you would like to be able to configure things I can figure out how to implement it.